subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/assignment/internal/KotlinDslAssignment.kt

```diff
@@ -16,10 +16,20 @@
 
 package org.gradle.kotlin.dsl.assignment.internal
 
+import org.gradle.internal.deprecation.DeprecationLogger
+
 
 /**
- * TODO: Remove once kotlin-dsl plugin is updated
+ * This class is used in `kotlin-dsl` plugin from 4.0.2 to 4.1.0. To be removed in Gradle 9.0.
  */
+@Suppress("unused")
 object KotlinDslAssignment {
-    fun isAssignmentOverloadEnabled() = true
+    fun isAssignmentOverloadEnabled(): Boolean {
+        DeprecationLogger.deprecateMethod(this::class.java, "isAssignmentOverloadEnabled()")
```

Comment:
❌ If we plan to remove the type we should emit a warning when that type is loaded saying that it will be removed.

Reply:
Sure we can do that, although this class is internal, so I wonder if we need to go to such extend to print something when class is loaded. 
 
It was used just on one place in `kotlin-dsl` where `isAssignmentOverloadEnabled`. So I think it makes sense to just check when this method is called.

Can you explain more why we should print a warning when type is loaded?

Reply:
My thinking is that we deprecate and will remove the type so why not warn on the type usage?

I don't see a reasonable reason for one to use that type but you never know.

Reply:
sure, we can add it then

Reply:
I would only have one deprecation for the type, then the deprecation for the method is not necessary and will just create noise.

Reply:
Makes sense

Reply:
Done